minifi/minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/command/CommandServiceFactory.java

```diff
@@ -0,0 +1,102 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.minifi.bootstrap.command;
+
+import java.io.File;
+import java.util.LinkedList;
+import java.util.List;
+import org.apache.nifi.minifi.bootstrap.MiNiFiParameters;
+import org.apache.nifi.minifi.bootstrap.RunMiNiFi;
+import org.apache.nifi.minifi.bootstrap.RunMiNiFiCommand;
+import org.apache.nifi.minifi.bootstrap.service.BootstrapFileProvider;
+import org.apache.nifi.minifi.bootstrap.service.CurrentPortProvider;
+import org.apache.nifi.minifi.bootstrap.service.GracefulShutdownParameterProvider;
+import org.apache.nifi.minifi.bootstrap.service.MiNiFiCommandSender;
+import org.apache.nifi.minifi.bootstrap.service.MiNiFiStatusProvider;
+import org.apache.nifi.minifi.bootstrap.service.MiNiFiStdLogHandler;
+import org.apache.nifi.minifi.bootstrap.service.PeriodicStatusReporterManager;
+
+public class CommandServiceFactory {
+
+    private final MiNiFiCommandSender miNiFiCommandSender;
+    private final CurrentPortProvider currentPortProvider;
+    private final MiNiFiParameters miNiFiParameters;
+    private final MiNiFiStatusProvider miNiFiStatusProvider;
+    private final PeriodicStatusReporterManager periodicStatusReporterManager;
+    private final BootstrapFileProvider bootstrapFileProvider;
+    private final MiNiFiStdLogHandler miNiFiStdLogHandler;
+    private final File bootstrapConfigFile;
+    private final RunMiNiFi runMiNiFi;
+    private final GracefulShutdownParameterProvider gracefulShutdownParameterProvider;
+
+    public CommandServiceFactory(MiNiFiCommandSender miNiFiCommandSender, CurrentPortProvider currentPortProvider, MiNiFiParameters miNiFiParameters,
+        MiNiFiStatusProvider miNiFiStatusProvider, PeriodicStatusReporterManager periodicStatusReporterManager,
+        BootstrapFileProvider bootstrapFileProvider, MiNiFiStdLogHandler miNiFiStdLogHandler, File bootstrapConfigFile, RunMiNiFi runMiNiFi,
+        GracefulShutdownParameterProvider gracefulShutdownParameterProvider) {
+        this.miNiFiCommandSender = miNiFiCommandSender;
+        this.currentPortProvider = currentPortProvider;
+        this.miNiFiParameters = miNiFiParameters;
+        this.miNiFiStatusProvider = miNiFiStatusProvider;
+        this.periodicStatusReporterManager = periodicStatusReporterManager;
+        this.bootstrapFileProvider = bootstrapFileProvider;
+        this.miNiFiStdLogHandler = miNiFiStdLogHandler;
+        this.bootstrapConfigFile = bootstrapConfigFile;
+        this.runMiNiFi = runMiNiFi;
+        this.gracefulShutdownParameterProvider = gracefulShutdownParameterProvider;
+    }
+
+    public CommandService getService(RunMiNiFiCommand command) {
```

Comment:
It doesn't make a big difference but you could create a map, pre-populate the different services at startup. Each service would have  a getCommand or something like that so just creating a new Command would automatically put it into that map in the right place and at runtime you would just get the proper service from the pre-populated map. Maybe it is slightly more decoupled that way but I'm totally fine with the current approach just wanted to note it.